### ISSUE TITLE
screenrant.com: breakage

### ISSUE THREAD

**Issue by Angelique465** [uBlockOrigin/uBOL-home#202]
UBolite version: 2024.9.1.1266 chromium

Issue Link: https://screenrant.com/videos/

All videos are blocked (complete, optimal & basic filtering modes)

**Comment by stephenhawk8054** [comment]
Hmm... It works on uBO. I don't see any incompatible filters for this site:

https://github.com/uBlockOrigin/uAssets/blob/b608854c4cff2d1a747bc14da5328754877bfd3b/filters/filters-2024.txt#L3019-L3030

**Comment by gorhill** [comment]
I can reproduce without any blocker extension enabled.

**Comment by gorhill** [comment]
Had to restart the tests to reproduce on my side. So far I can reproduce an issue which is caused by the `##.video` cosmetic filter. The issue does not reproduce when using _Basic_ mode.

This exception filter is not being enforced in uBOL: `screenrant.com#@#.video`, so similar issue as with <https://github.com/uBlockOrigin/uBOL-home/issues/181>.

**Comment by Angelique465** [comment]
In Basic mode, when you click on some video links on the right (Trending) some of them are not played

**Comment by stephenhawk8054** [comment]
Hmm... I thought https://github.com/uBlockOrigin/uBOL-home/issues/181 is about exceptions for generic cosmetic filters. Didn't know exceptions for specific cosmetic filters have issues too.

**Comment by Yuki2718** [comment]
IIRC we didn't agree on whether to block or not videos like https://github.com/easylist/easylist/commit/39dc96c1df7436cfb21c941a34965db93d3abf0a I don't see the breakage on `https://screenrant.com/videos/`,

**Comment by stephenhawk8054** [comment]
I actually didn't see `#@#.video` being logged in uBO logger with OP's link, but I haven't checked again yet.

@gorhill Do you see the filter being reported in uBO?

**Comment by Yuki2718** [comment]
It's by these filters
```
##.adsninja-ad-zone
##.w-adsninja-video-player
##.ad-zone:not(.textads)
##.ad-zone-container
##.ad-current
```

**Comment by stephenhawk8054** [comment]
> It's by these filters
> ```
> ##.adsninja-ad-zone
> ##.w-adsninja-video-player
> ##.ad-zone:not(.textads)
> ##.ad-zone-container
> ##.ad-current
> ```

I think they should be defused by `ghide` already?

**Comment by Yuki2718** [comment]
> I think they should be defused by `ghide` already?

In uBO this is the case, IDK why not on uBOL.

**Comment by Yuki2718** [comment]
Probably the real cause will be `||securepubads.g.doubleclick.net/tag/js/gpt.js$script,redirect=googletagservices_gpt.js:5`. Does uBOL support priority with redirect?

**Comment by gorhill** [comment]
> Do you see the filter being reported in uBO?

Yes:

![image](https://github.com/user-attachments/assets/50f5d0d5-feb6-473c-8097-b7368f355b0c)

Specific URL used to reproduce the issue: `https://screenrant.com/video/its-always-sunny-in-philadelphia-needs-to-end/`

I get a mostly blank page in uBOL.

Looking at the code again, specific exception filters are not enforced at all, regardless of whether it's generic or specific cosmetic filtering.

**Comment by stephenhawk8054** [comment]
@gorhill Yes, the videos inside each article actually need the `.video` exceptions. However, the one I'm confusing is OP's link in which I don't see any `.video` elements in the DOM.

**Comment by gorhill** [comment]
Right, I don't have cosmetic filtering issue at `https://screenrant.com/videos/`. I cannot reproduce the other issue _"when you click on some video links on the right (Trending) some of them are not played"_ at that URL on my side.

**Comment by stephenhawk8054** [comment]
> Right, I don't have cosmetic filtering issue at `https://screenrant.com/videos/`

Yeah, that's what I don't understand why the issue occurs with that link even though I only see network and scriptlet filters being logged in uBO. The blocked and redirected links look the same when I compared both network devtools.

**Comment by Angelique465** [comment]
> > In Basic mode, when you click on some video links on the right (Trending) some of them are not played
> 
> I cannot reproduce. You say _some of them_, so if this affects specific videos, we will need more details about which ones so that we can try to reproduce on our side.
> 
> Also, please disclose the lists you enabled in uBOL -- this is an important configuration detail.

https://app.screencast.com/Tp4vWj8HobBQy

Ther's no particular settings in UBol, installed and used as it is.

**Comment by stephenhawk8054** [comment]
I'm scratching my head now. I added these

```adb
@@||adsninja.ca^
@@*$ehide,domain=com|net|ca|org|io|tech|gt|ad.gt|com.vn|mx|me
@@*$3p,1p,to=com|net|ca|org|io|tech|gt|ad.gt|com.vn|mx|me
com,net,ca,org,io,tech,gt,ad.gt,com.vn,mx,me#@#+js()
```

to both `ublockorigin.github.io_uAssets_thirdparties_easylist.txt` and `ublockorigin.github.io_uAssets_filters_filters.min.txt` but the video in `https://screenrant.com/videos/` still doesn't appear. Is `ehide` available on uBOL? If not, how can I disable all cosmetic filtering in a site?

**Comment by stephenhawk8054** [comment]
I think I found the issues. It's caused by these 2 filters:

```adb
~howtogeek.com,~makeuseof.com,~xda-developers.com##.ad-zone:not(.textads)

androidpolice.com,cbr.com,collider.com,dualshockers.com,gamerant.com,howtogeek.com,makeuseof.com,movieweb.com,screenrant.com,thegamer.com,xda-developers.com##+js(set, Object.prototype.ads.nopreroll_, true)
```

in which the 1st is the one hiding the video (the scriptlet one is just causing video unable to play, not hiding video player). I guess

```adb
@@*$ghide,domain=androidpolice.com|cbr.com|collider.com|dualshockers.com|gamerant.com|howtogeek.com|movieweb.com|makeuseof.com|screenrant.com|thegamer.com|xda-developers.com
```

can't affect `~howtogeek.com,~makeuseof.com,~xda-developers.com##.ad-zone:not(.textads)` ?

**Comment by stephenhawk8054** [comment]
So, about `##.video` situation, how can we solve the issue if EL does not remove the filter `ustthenews.com,movieweb.com,screenrant.com,thegamer.com##.video` ?

**Comment by Yuki2718** [comment]
I can reproduce full breakage on `https://screenrant.com/video/its-always-sunny-in-philadelphia-needs-to-end/` so at least will remove `screenrant.com##.video`

**Comment by stephenhawk8054** [comment]
@Yuki2718 These sites have the same breakage so I think they can be removed too

```yml
https://movieweb.com/video/depravity-movie-trailer/
https://www.thegamer.com/video/someone-should-make-a-game-that-s-all-character-creator/
```

---

@gorhill Sorry, so just to confirm, `~example.com##.ads` will be considered as specific cosmetic filter and `ghide` won't work for other sites except `example.com` right?

**Comment by Yuki2718** [comment]
I think `~example.com##.ads` is a generic filter.

**Comment by stephenhawk8054** [comment]
Yeah, `ghide` works in uBO with `~example.com##.ads` but somehow it doesn't work on uBOL, so I'm not sure if that filter is assessed differently on uBOL or not.

**Comment by stephenhawk8054** [comment]
The broken link is `https://screenrant.com/videos/`. This one doesn't have `.video` elements in the page.

**Comment by stephenhawk8054** [comment]
It's not `.textads`. The item being hidden is `.ad-zone:not(.textads)`

**Comment by gorhill** [comment]
Oops ok. Still, I am really confused here, as I am trying to investigate `@@...ghide` not working for generic cosmetic filters -- I cannot think of why this would not work for `~howtogeek.com,~makeuseof.com,~xda-developers.com##.ad-zone:not(.textads)`.

I also cannot find instance of elements matching `.ad-zone` at either `https://screenrant.com/videos/` or `https://screenrant.com/video/its-always-sunny-in-philadelphia-needs-to-end/`.

**Comment by stephenhawk8054** [comment]
Hmm... Weird, I can see it with uBO picker in `https://screenrant.com/videos/`

![image](https://github.com/user-attachments/assets/00aaae19-928e-4e7f-9e29-42f44b1e525e)

Devtools also shows it

![image](https://github.com/user-attachments/assets/ab454105-766a-4a04-9324-e85a410fb7e8)

**Comment by gorhill** [comment]
Looks like I am being served a different DOM:

![image](https://github.com/user-attachments/assets/f757d1aa-2756-4d4f-81b6-964dde137bea)


Using default lists, etc. Same result when I tried through a U.S. VPN.

---

Hmm... "No-Ads mode"?

![image](https://github.com/user-attachments/assets/e0bcc0a6-23a1-4888-a9f7-4cf46e63659b)

**Comment by stephenhawk8054** [comment]
@gorhill Can you try this example:

- Build uBOL locally and add these filters manually (to `ublockorigin.github.io_uAssets_filters_filters.min.txt` file)

```adb
~github.com##.post-content
@@||rpgbot.net^$ghide
```

- Access `https://rpgbot.net/video-games/baldurs-gate-3/spells/minor-illusion/`
- The `.post-content` element is hidden on my side

<details>

![image](https://github.com/user-attachments/assets/7476345b-9d35-40ef-9b93-18c11f3fd99e)


</details>

**Comment by gorhill** [comment]
Right, I meant to test with `~toto.com##body` and `@@||example.com^$ghide` then I forgot. I can reproduce with these filters.

**Comment by Angelique465** [comment]
uBlock Origin Lite version 2024.9.12.1004 Chromium
Settings: Basic | Optimal | Complete
Link: https://screenrant.com/videos/

Videos are still intermittently hided. To show and play a CTRL-F5 is needed (I previously clean the cache in browser)

See this video: https://app.screencast.com/9MwsMaq8CerdJ

**Comment by stephenhawk8054** [comment]
I can't reproduce with `2024.9.12.1004`

![image](https://github.com/user-attachments/assets/b18c75fa-060f-42c6-9814-179cab0de9a6)

**Comment by Angelique465** [comment]
uBlock Origin Lite version 2024.9.12.1004 Firefox version 130.0 (64bit) official
Settings: Basic | Optimal | Complete
Link: https://screenrant.com/videos/

I can confirm it works fine in Firefox, although FF refuses to install the XPI file (not signed component) so I had to install it manually via about://debugging -> This Firefox.  But the option page is not automatically shown and UBol starts in Complete mode. Is it normal?

**Comment by stephenhawk8054** [comment]
Ok in chromium I need

```adb
@@||a.ad.gt/api/v1/u/matches/$script,domain=screenrant.com
@@||cdn.adsninja.ca/adsninja_worker$script,domain=screenrant.com
```

**Comment by stephenhawk8054** [comment]
@Angelique465 Can you test this version: https://github.com/stephenhawk8054/uBOL-test/releases/tag/uBOLite_test.chromium_2024.9.13.622 ?

**Comment by Angelique465** [comment]
uBlock Origin Lite version 2024.9.12.1004 Chromium
Link: https://screenrant.com/videos/ and child links

Cache clean
Still videos are intermittently hided. Sometimes refreshing the page allows the video to be shown, sometimes refreshing the page (when a video is normally shown), hides the video again.
CTRL-F5 normally always shows hidden videos
Something regarding cache?

https://app.screencast.com/drvfj1d5sdah9

**Comment by stephenhawk8054** [comment]
Hmm... Ok I suspect this is service worker issue

```adb
||screenrant.com^$csp=worker-src 'none'
```

**Comment by Angelique465** [comment]
No screenrant.com in `chrome://serviceworker-internals`

**Comment by stephenhawk8054** [comment]
@Angelique465 Can you test this: https://github.com/stephenhawk8054/uBOL-test/releases/tag/uBOLite_test.chromium_2024.9.13.749

**Comment by Angelique465** [comment]
IT WORKS!!! 
@stephenhawk8054 Chapeau!
